src/util/entities/Channel.ts

```diff
@@ -243,6 +262,100 @@ export class Channel extends BaseClass {
 		return channel;
 	}
 
+	static async createThreadChannel(
+		channel: Partial<Channel>,
+		metadata: Partial<ThreadMetadataSchema>,
+		user_id: string = "0",
+		opts?: {
+			keepId?: boolean;
+			skipExistsCheck?: boolean;
+			skipParentExistsCheck?: boolean;
+			skipPermissionCheck?: boolean;
+			skipEventEmit?: boolean;
+			skipNameChecks?: boolean;
+		}
+	): Promise<Channel> {
+		channel = {
+			// set the default type to private
+			type: ChannelType.GUILD_PRIVATE_THREAD,
+			...channel,
+			...(!opts?.keepId && { id: Snowflake.generate() }),
+			created_at: new Date(),
+			position: 0, // TODO:
+			message_count: 0,
+			member_count: 1,
+			total_message_sent: 0
+		};
+
+		const exists = await Channel.findOne({
+			where: {
+				id: channel.id
+			}
+		});
+		if (!opts?.skipExistsCheck && exists) throw DiscordApiErrors.THREAD_ALREADY_CREATED_FOR_THIS_MESSAGE;
```

Comment:
add a guild feature to disable "thread already existing for starting message" check

Reply:
added as `ALLOW_EXISTING_THREAD_FOR_MESSAGE`